SimCalorimetry/HGCalSimAlgos/interface/HGCalRadiationMap.h

```diff
@@ -7,36 +7,38 @@
 #include "vdt/vdtMath.h"
 #include <string>
 
-typedef std::array<double, 8> radiiVec;
-
 /**
    @class HGCalRadiationMap
    @short parses a txt file with dose/fluence parameters and provides functions for noise, etc.
  */
 class HGCalRadiationMap {
 public:
   struct DoseParameters {
-    DoseParameters() : a_(0.), b_(0.), c_(0.), d_(0.), e_(0.), f_(0.), g_(0.), h_(0.), i_(0.), j_(0.) {}
-    double a_, b_, c_, d_, e_, f_, g_, h_, i_, j_;
+    DoseParameters()
+        : a_(0.), b_(0.), c_(0.), d_(0.), e_(0.), doff_(0.), f_(0.), g_(0.), h_(0.), i_(0.), j_(0.), foff_(0.) {}
+    double a_, b_, c_, d_, e_, doff_, f_, g_, h_, i_, j_, foff_;
   };
 
-  HGCalRadiationMap(){};
+  HGCalRadiationMap();
   ~HGCalRadiationMap(){};
 
   typedef std::map<std::pair<int, int>, DoseParameters> doseParametersMap;
 
   void setGeometry(const CaloSubdetectorGeometry *);
   void setDoseMap(const std::string &, const unsigned int);
 
-  double getDoseValue(const int, const int, const radiiVec &, bool logVal = false);
-  double getFluenceValue(const int, const int, const radiiVec &, bool logVal = false);
+  double computeRadius(const HGCScintillatorDetId &);
+
+  double getDoseValue(const int, const int, const double &, bool logVal = false);
```

Comment:
What is the rationale of passing a `const double &`?
The `&` was there before since the passed type was an `std::array`.

Reply:
ok